applications/dot_marketplace-phase2.md

```diff
@@ -0,0 +1,218 @@
+* **Costs:**  8,325 USD  
+
+The main deiverable for this milestone will be to migrate the existing application to substrate frame v2 and include the chat feature as a pallet for communication between a customer and a service provider to have a one-on-one conversation over the deliverables and timelines. The entire milestone covers the Rust/Substrate code implementation.
+
+| Sr no. | Deliverable | Description |
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Migration from frame v1 to frame v2 | The [existing application backend](https://github.com/WowLabz/tasking_backend/blob/Phase1_Milestone3/pallets/pallet-tasking/src/lib.rs) will be migrated to frame v2 to account for the new features provided by the framework |
+| 2 | Chat Pallet | Chat functionality is to be exposed and consumed between the customer and the service provider to ease communication and this will be integrated with the tasking pallet's frame v2 version |
+|  |  |  |
+
+### **Milestone 2**
+
+* **Estimated duration:** 2 weeks
+* **FTE:**  2
+* **PTE:**  2
+* **Costs:**  15,900 USD  
+
+
+In continuation to previous work, this milestone involves creation of a on chain decentralized court to handle dispute resolution. Being a juror is one of the user incentives that can be achieved thanks to the rating module mentioned in the first phase of dot marketplace. The entire milestone covers the Rust/Substrate code implementation.
+
+| Sr no. | Deliverable | Description |
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Decentralized Court Module | An on chain decentralized court for dispute resolution within the ecosystem. |
+| 1a | Disapprove Task  | In the case of a customer not being satisfied by the work submitted by the service provider (worker). A set of jurors is formed (court-summon) to resolve the dispute and pass a verdict. |
+| 1b | Disapprove Rating | The customer or the service provider, once they have received their rating for a particular task and are not satisfied by it. |
+| 1c | General Dispute | A general dispute function for cases that do not fall under the categories mentioned in 1a and 1b. |
```

Comment:
I took a look at the Tasking Workflow Diagram where you show that the Court process can be started from 3 different starting points:

1. Approve Task by Customer: No
2. Worker satisfied by rating: No
3. Customer satisfied by rating: No

I would assume that the court process references these starting points wherever there is a dispute, is that correct? Hence, the dispute logic for 1a-1c would be the same, is that correct? I'm a bit confused why you'd implement an additional dispute function for 1c. Please feel free to correct me if I'm wrong.

Reply:
Yes, you are correct regarding the court process about being called at any one of these instances. The dispute logic is the same throughout the calls. The only difference is the scenarios when the dispute is called. The dispute logic is a function which is called from two extrinsics. The additional dispute function for 1c is to cover one case (at the moment), where lets say the customer is not closing the task due to some reason. The worker gets the functionality to raise dispute for this particular case. Since we thought there can be more cases as such, why not create an extrinsic which only raises a dispute.

Reply:
Thanks for explaining, sounds good.
Maybe it would make sense to add this scenario to the application as well.

Reply:
@varun24143 thanks for [adding this](https://github.com/w3f/Grants-Program/pull/832/files#diff-cbf2c9be64c2402648e5f355330c8ca6221cf239b8bfcaf95e67da105bd66062R179) to the doc. Feel free to add a reference/link to your comment in the future, this makes it easier for us to review what exactly got changed and might speed up the review process.